.claude/hooks/pre-commit.sh

```
@@ -1,39 +1,83 @@
+STAGED_GO=$(git diff --cached --name-only --diff-filter=ACMR -- '*.go' 'go.mod' 'go.sum')
+
+if [ -z "$STAGED_GO" ]; then
   echo "Pre-commit: only non-Go files staged, skipping lint and tests." >&2
   exit 0
 fi
 
+KEY=""
+if [ "${PRECOMMIT_NO_CACHE:-0}" != "1" ] && [ -f tools/buildcache/buildcache.sh ]; then
+  # shellcheck disable=SC1091
+  source tools/buildcache/buildcache.sh
+  KEY=$(buildcache_key $STAGED_GO)
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)